proposals/credential-management.md

```diff
@@ -0,0 +1,128 @@
+
+First, the registerContainer function allows you to register your extensions implementation of ExtensionCredentialsContainer interface.
+
+Request:
+
+```js
+const credentialsContainer = {
+    create(options) {
+        console.log("Demo Extension: credentials.create");
+        return Promise.resolve(null);
+    },
+    get(options) {
+        console.log("Demo Extension: credentials.get");
+        return Promise.resolve(null);
+    },
+    store(credential) {
+        console.log("Demo Extension: credentials.store");
+        return Promise.resolve();
+    }
+}
+
+browser.credentials.registerContainer(credentialsContainer);
+```
+
+**browser.credentials.isActiveContainer**
+
+This retrieves a boolean indicating if this extension is the active credential manager.
+
+```js
+browser.credentials.isActiveContainer();
+```
+
+## Manifest
+
+To make use of the browser.credentials namespace a new `credentials` key would be required inside of the extension manifest.
+
+```json
+{
+  "credentials": {
+    "supportedTypes": [
+      "password",
+      "publicKey"
+    ]
+  }
+}
+```
+
+## FAQ
+
+**Why is it useful to allow supporting multiple credential types?**
+
+The majority of websites today that use authentication use passwords. However, in the future sites will make use of new technology such as passkeys (PublicKey credentials), and password manager extensions will want to support these new credentials.
+
+**Are there any alternatives?**
+
+Today password manager extensions use a signals such as heuristics for detecting login forms as well as signals such as a page navigation to determine successful login. Along with overriding of navigator.credentials to intercept requests for Passkeys.
+
+The first one is susceptible to false results, single page applications lack the conventional navigation on success. Some sites also navigate on error which extensions could misinterpret as a success.
+
+**Are there any extensions that would use this?**
+
+Any conventional password manager extension would benefit from this API. However, those looking to support Passkeys would particularly benefit.
+
+The Dashlane browser extension, a password manager with over 4M weekly active users in the Chrome Web Store, currently provides functionality to create and retrieve Passkey credentials.
+
+The 1Password password manager also looks set to support Passkeys soon.
+
+**Would this API encourage use of passwords which are bad for security?**
+
+While it could be seen to be encouraging password authentication, this API (and its Web API sibling) on its own wouldn't change the fact that the vast majority of websites use passwords.
+This API would help to encourage best practices such as strong, unique per site passwords by making them more ergonomic to use.
+
+This API would also help extensions and in turn websites embrace the passwordless future.
+
+**Only Chromium implements the password credential type. Would this limit the usefulness of this API?**
```

Comment:
This to me is this biggest blocker to the usefulness of this API without some sort of "custom" credentials support.